nixos/modules/services/misc/weechat.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.weechat;
+in
+
+{
+  options.services.weechat = {
+    enable = mkEnableOption "weechat";
+    init = mkOption {
+      description = "Weechat commands applied at start, one command per line.";
+      example = ''
+        /set relay.network.password correct-horse-battery-staple
+        /relay add weechat 9001
+      '';
+      type = types.str;
+      default = "";
+    };
+    root = mkOption {
+      description = "Weechat state directory.";
+      type = types.str;
+      default = "/var/lib/weechat";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users = {
```

Comment:
I commonly come across shell-hosts that offer a bunch of weechat sessions (usually one per person, user, …).

Would you think it could make sense to refactor this to allow multiple weechat sessions per host?

Reply:
There are quite a few NixOS services that could benefit from running multiple copies. Tor, for example, is one of them, if someone is running an exit node and has multiple IPv4 addresses, it would make sense to run one instance of Tor per IPv4 address.

I think this should be implemented through containers.

Reply:
Good point. Probably a sensible thing to do then 👍